src/graphcast_agent/mod.rs

```diff
@@ -47,6 +49,87 @@ pub const REGISTRY_SUBGRAPH: &str =
 /// A constant defining the goerli network subgraph endpoint.
 pub const NETWORK_SUBGRAPH: &str = "https://gateway.testnet.thegraph.com/network";
 
+#[derive(Debug, thiserror::Error)]
+pub enum ConfigError {
+    #[error("Validate the input: {0}")]
+    ValidateInput(String),
+    #[error("Unknown error: {0}")]
+    Other(anyhow::Error),
+}
+
+pub async fn validate_set_up(
```

Comment:
Could this function be an impl of the GraphcastAgentConfig struct?

Reply:
indeed, good idea